src/os/ui/menu/listmenu.js

```diff
@@ -193,6 +215,46 @@ os.ui.menu.list.onSortSelected = function(event) {
 };
 
 
+/**
+ * Handle the "Color Selected" menu event.
+ *
+ * @param {os.ui.menu.MenuEvent} event The menu event.
+ */
+os.ui.menu.list.onColorSelected = function(event) {
+  var context = event.getContext();
+  if (os.instanceOf(context, os.source.Vector.NAME)) {
+    var source = /** @type {!os.source.Vector} */ (context);
+    if (source && source.getSelectedItems()) {
+      var items = source.getSelectedItems();
+
+      // call the confirm window from this context so it doesn't appear in the wrong tab
+      os.ui.window.launchConfirmColor(function(color) {
+        source.setColor(items, os.style.toRgbaString(color));
+      }, os.feature.getFirstColor(items));
+    }
+  }
+};
+
+
+/**
+ * Handle the "Color Selected" menu event.
+ *
+ * @param {os.ui.menu.MenuEvent} event The menu event.
+ */
+os.ui.menu.list.onResetColor = function(event) {
+  var context = event.getContext();
+  if (os.instanceOf(context, os.source.Vector.NAME)) {
+    var source = /** @type {!os.source.Vector} */ (context);
+    if (source && source.getColorModel()) {
+      var cm = source.getColorModel();
+      if (cm) {
+        cm.setColorMethod(os.data.histo.ColorMethod.RESET);
+        cm.setColorMethod(os.data.histo.ColorMethod.MANUAL);
```

Comment:
Calling both of these back to back seems odd to me. Should the reset call just set it back to manual automatically?

Reply:
Yes and no.  I made some other changes that OBE'd this requirement (it used to have to re-enable the manual colors to pick up the colors off the features).  I put this (and the others) back to only resetting.